crates/valence_equipment/src/lib.rs

```diff
@@ -0,0 +1,317 @@
+use bevy_app::prelude::*;
+use bevy_ecs::prelude::*;
+#[cfg(feature = "inventory")]
+pub use inventory_sync::EquipmentInventorySync;
+use valence_server::client::{Client, FlushPacketsSet, LoadEntityForClientEvent};
+use valence_server::entity::living::LivingEntity;
+use valence_server::entity::{EntityId, EntityLayerId, Position};
+use valence_server::protocol::packets::play::entity_equipment_update_s2c::EquipmentEntry;
+use valence_server::protocol::packets::play::EntityEquipmentUpdateS2c;
+use valence_server::protocol::WritePacket;
+use valence_server::{EntityLayer, ItemStack, Layer};
+
+pub struct EquipmentPlugin;
+
+impl Plugin for EquipmentPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
+            PreUpdate,
+            (
+                on_entity_init,
+                #[cfg(feature = "inventory")]
+                inventory_sync::on_attach_inventory_sync,
+            ),
+        )
+        .add_systems(
+            Update,
+            (
+                update_equipment.before(FlushPacketsSet),
+                on_entity_load.before(FlushPacketsSet),
+                #[cfg(feature = "inventory")]
+                inventory_sync::equipment_inventory_sync,
```

Comment:
We don't need to put this behind a feature, the system will not run if nothing matches the query.